gslib/gcs_json_api.py

```diff
@@ -244,15 +244,15 @@ def __init__(self,
     # sequentially, but we can share TCP warmed-up connections across calls.
     self.download_http = self._GetNewDownloadHttp()
     self.upload_http = self._GetNewUploadHttp()
+    WrapDownloadHttpRequest(self.download_http)
```

Comment:
Doing it above due to the mismatch of return type of authorize method in p12 credentials, json key.